pkg/kubelet/winstats/winstats.go

```diff
@@ -0,0 +1,311 @@
+// +build windows
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package winstats provides a client to get node and pod level stats on windows
+package winstats
+
+import (
+	"context"
+	"encoding/json"
+	dockerstatstypes "github.com/docker/docker/api/types"
+	dockerapi "github.com/docker/engine-api/client"
+	dockertypes "github.com/docker/engine-api/types"
+	cadvisorapi "github.com/google/cadvisor/info/v1"
+	cadvisorapiv2 "github.com/google/cadvisor/info/v2"
+	"k8s.io/kubernetes/pkg/kubelet/network"
+	"os"
+	"runtime"
+	"sync"
+	"time"
+)
+
+// Client is an object that is used to get stats information.
+type Client struct {
+	dockerClient                *dockerapi.Client
```

Comment:
Adding more docker-specific code in the main kubelet codebase is a big no to me.